### ISSUE TITLE
Use repository on the local frontend filesystem directly

### ISSUE THREAD

**Issue by pokrakam** [abapGit/abapGit#3669]
It would be useful to interact with a local repo on the SapGUI machine. The closest we have is up/download of a zip file, but it would be nicer if one could directly use a real repo located at e.g. /users/mike/git/abapGit on my laptop.

It's possible to set up a git server on the local machine using third party software, and I've tried to set up an Apache server to serve up my local repo (unsuccessful so far, but I haven't spent too much time on it). But I thought I'd put the idea out here to see whether it's been discussed, if it's reasonably doable, or just a crazy idea that nobody else is interested in :-)

Why? There are many reasons to favour a local / clone repo over a GitHub one. Perhaps legal (data not permitted outside company servers), or simply practical because it's easier to navigate histories or do branch manipulation or do comparisons or just about any other detailed git operations on a local repo using either the command line or GUI tools such as VSCode, Eclipse, etc.

**Comment by larshp** [comment]
install https://docs.gitea.io/en-us/install-with-docker/ or https://docs.gitlab.com/omnibus/docker/ or one of the other free alternatives

alternatively https://github.com/larshp/abapGitServer

**Comment by pokrakam** [comment]
Thanks, am using abapGitServer, and from what I can see the others you suggest look similar in that they are a third party option where I'd have to set up a repo only accessible via http.

AGS is great, but does still have a couple of minor limitations (and a major limitation of available time for me to do much about it). Currently I push-pull to/from AGS via the command line or Eclipse and it works well. So one of my scenarios is:
local repo <-> AGS <-> AG/SAP

Where it gets messy is if I want to keep a fork (either on GitHub or AGS) in sync with the original. Even GitHub suggests going via a local repo to do this at https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

This is essentially a clumsy three-repo `checkout-pull-fetch-rebase-push` merge, four repos if you count the AG one. 

So if my fork is in AGS and I want to synch it up, I must: 
```
<AG push>
$ git checkout master
$ git pull ags
$ git fetch upstream
$ git rebase upstream/master
<merge with my working branch>
$ git push ags
<AG pull>
```

If abapGit can use the filesystem repo then that becomes my main fork and  we eliminate one layer. A sync process becomes:
```
<AG export/push>
$ git pull
<merge with my working branch>
<AG pull>
```

Aside from that, I like using terminal. It's fast, none of that web wait. My repo is right there in my git folder. I can edit it from my file browser or any editor I like. It gets backed up. If I mess up my repo (noooo, never done that before), I can restore it from my last filesystem snapshot. 
So I don't know how many others this relates to, maybe I'm the only one ;-) But I thought I'd post it to see what the responses are. It's certainly not a dealbreaker, I've got processes that work. I just figured that the ZIP module already does a large part of it so it may or may not be an easy thing to implement (must admit haven't looked into it myself).
